dali/operators/image/resize/resize_attr.cc

```diff
@@ -0,0 +1,433 @@
+      }
+    } else {
+      // NotLarger or NotSmaller mode
+
+      // First, go through all the dimensions that have their scale defined and find
+      // the min (or max) scale
+      double final_scale = 0;
+      bool first = true;
+      for (int d = 0; d < spatial_ndim_; d++) {
+        if (mask[d]) {
+          float s = std::abs(scale[d]);
+          if (first ||
+              (mode_ == ResizeMode::NotSmaller && s > final_scale) ||
+              (mode_ == ResizeMode::NotLarger && s < final_scale))
+            final_scale = s;
+          first = false;
+        }
+      }
+
+      // If there's a size limit, apply it and possibly reduce the final scale
+      if (has_max_size_) {
+        for (int d = 0; d < spatial_ndim_; d++) {
+          if (max_size_[d] > 0) {
+            double s = static_cast<double>(max_size_[d]) / in_size[d];
+            if (s < final_scale)
+              final_scale = s;
+          }
+        }
+      }
+
+      // Now, let's apply the final scale to all dimensions - if final_scale is different (in
+      // absolute value) than one defined by the caller, adjust it; also, if no scale was defined
+      // for a dimension, then just use the final scale.
+      for (int d = 0; d < spatial_ndim_; d++) {
+        if (!mask[d]) {
+          out_size[d] = in_size[d] * final_scale;
+          scale[d] = final_scale;
+        } else if (std::abs(scale[d]) != final_scale) {
+          scale[d] = std::copysign(final_scale, scale[d]);
+          out_size[d] = in_size[d] * scale[d];
+        }
+      }
+    }
+  }
+}
+
+void ResizeAttr::CalculateSampleParams(ResizeParams &params,
+                                       SmallVector<float, 3> requested_size,
+                                       SmallVector<float, 3> in_lo,
+                                       SmallVector<float, 3> in_hi,
+                                       bool adjust_roi) {
+  assert(static_cast<int>(requested_size.size()) == spatial_ndim_);
+  assert(static_cast<int>(in_lo.size()) == spatial_ndim_);
+  assert(static_cast<int>(in_hi.size()) == spatial_ndim_);
+
+  for (int d = 0; d < spatial_ndim_; d++) {
+    DALI_ENFORCE(in_lo[d] != in_hi[d] || requested_size[d] == 0,
+                "Cannot produce non-empty output from empty input");
+  }
+
+  SmallVector<float, 3> in_size;
+  in_size.resize(spatial_ndim_);
+  for (int d = 0; d < spatial_ndim_; d++) {
+    float sz = in_hi[d] - in_lo[d];
+    if (sz < 0) {
+      std::swap(in_hi[d], in_lo[d]);
+      requested_size[d] = -requested_size[d];
+      sz = -sz;
+    }
+    in_size[d] = sz;
+  }
+
+  AdjustOutputSize(requested_size.data(), in_size.data());
+
+  params.resize(spatial_ndim_);
+  params.src_lo = in_lo;
+  params.src_hi = in_hi;
+
+  for (int d = 0; d < spatial_ndim_; d++) {
+    float out_sz = requested_size[d];
+    bool flip = out_sz < 0;
+    params.dst_size[d] = std::max(1, round_int(std::fabs(out_sz)));
+    if (flip)
+      std::swap(params.src_lo[d], params.src_hi[d]);
```

Comment:
If lo was  > hi, and you swap them here. How does the flipping work?

Reply:
There are two ways to flip:
1. negative output size
2. lo > hi
The final result uses the latter, but AdjustOutputSize uses the former (hence the first swap).